sdk/lib/core/uri.dart

```diff
@@ -512,21 +554,38 @@ abstract interface class Uri {
 
   /// The fragment identifier component.
   ///
+  /// A fragment identifier is a string after URI, after the hash, which
+  /// identifies something specific as a function of the document. For
+  /// a user interface Web document such as HTML page, it traditionally
+  /// identifies a section or part of the page. In modern pages, it can
+  /// indicate a particular view of the page.
+  ///
   /// The value is the empty string if there is no fragment identifier
   /// component.
   String get fragment;
 
-  /// The URI path split into its segments.
+  /// Same as the URI [path], but split into its segments.
+  ///
+  /// This is an alternative way to access the [path] component and
+  /// is useful for when a program needs to access components of a URI
+  /// such as the immediate parent folder of the current file.
+  /// This kind of field is valuable for extracting folder names
```

Comment:
I'd drop the "kind of field", it's not clear what it means. The folder names was mentioned jsut above.

So maybe just
```dart
  /// Where the [path] contains the URIs path component, including any escapes,
  /// the individual segments of `pathSegments` have been unescaped and can,
  /// for example, easily be displayed in a breadcrumb display.
```
```
I'd proably just drop it. I wouldn'